metadata-db-app/int_test/cmr/metadata_db/int_test/concepts/service_save_test.clj

```diff
@@ -1,37 +1,101 @@
 (ns cmr.metadata-db.int-test.concepts.service-save-test
-  "Contains integration tests for saving services. Tests saves with various configurations including
-  checking for proper error handling."
+  "Contains integration tests for saving services. Tests saves with various
+  configurations including checking for proper error handling."
   (:require
    [clojure.test :refer :all]
    [cmr.common.util :refer (are3)]
    [cmr.metadata-db.int-test.concepts.concept-save-spec :as c-spec]
    [cmr.metadata-db.int-test.utility :as util]))
 
-;;; fixtures
-;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Fixtures & one-off utility functions
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
-(use-fixtures :each (util/reset-database-fixture {:provider-id "REG_PROV" :small false}))
+(use-fixtures :each (util/reset-database-fixture
+                     {:provider-id "PROV1" :small false}
+                     {:provider-id "PROV2" :small false}))
 
 (defmethod c-spec/gen-concept :service
   [_ _ uniq-num attributes]
-  (util/service-concept uniq-num attributes))
-
-;; tests
-;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
-
-;; CMR-4335 - services
-;; This seems to fail because we keep creating new revisions of the same concept over and over.
-; (deftest save-service-test
-;   (c-spec/general-save-concept-test :service ["CMR"]))
-
-(deftest save-service-specific-test
-  (testing "saving new services"
-    (are3 [service exp-status exp-errors]
-          (let [{:keys [status errors]} (util/save-concept service)]
-            (is (= exp-status status))
-            (is (= (set exp-errors) (set errors))))
-
-          "failure when using non system-level provider"
-          (assoc (util/service-concept 2) :provider-id "REG_PROV")
-          422
-          ["Service could not be associated with provider [REG_PROV]. Services are system level entities."])))
+  (util/service-concept "PROV1" uniq-num attributes))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Tests
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+ (deftest save-service
+   (c-spec/general-save-concept-test :service ["PROV1"]))
+
+(deftest save-service-with-missing-required-parameters
+  (c-spec/save-test-with-missing-required-parameters
+   :service ["PROV1"] [:concept-type :provider-id :native-id :extra-fields]))
+
+(deftest save-service-created-at
```

Comment:
Looks like a copy paste of another test. Might be worth trying to generalize to work with any concept type now that we have quite a few concept types.

Reply:
Okay, there were two tests essentially duplicated in services and variables test suites. I've generalized these in `cmr.metadata-db.int-test.utility` and have updated the tests to use the utility functions now, instead.